### PR TITLE
Enable predefined IPs for all cluster nodes

### LINKED ISSUE #191: Use external LB and DNS for install
We should have a way to disable haproxy and DNS in bastion when using external LB and DNS services.

### PR DESCRIPTION
**Pull request by cs-zhang**
Fixes #191

enable to use predefined IPs for all cluster nodes

Signed-off-by: CS Zhang <zhangcho@us.ibm.com>

### CODE REVIEW

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/201#" title="Author self-approved">cs-zhang</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **yussufsh** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @yussufsh` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervm).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by Prajyot-Parab — APPROVED** [review]
/lgtm

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/201#" title="Author self-approved">cs-zhang</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/201#pullrequestreview-653013382" title="LGTM">Prajyot-Parab</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **yussufsh** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @yussufsh` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervm).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: modules/3_helpernode/helpernode.tf
Comment (yussufsh) [inline modules/3_helpernode/helpernode.tf]: This is intentional? I guess helpernode node does not support these vars yet.

File: variables.tf
Comment (yussufsh) [inline variables.tf]: These should not be part of this commit.

**Review by yussufsh — CHANGES_REQUESTED** [review]
@cs-zhang Can you remove dns and lb params from this commit? This should be a different PR when helpernode support them.

Another thing, from code it is allowing me to optionally set only master fixed IPs, we could ignore worker,bootstrap fixed IPs. This will lead to the race condition and block the reserved IP. In other words, workers who can take dynamic IP may pick up the fixed IP for master from the subnet. See https://github.com/ocp-power-automation/ocp4-upi-powervm/issues/182 where we faced similar issue.

File: modules/3_helpernode/helpernode.tf
Comment (cs-zhang) [reply modules/3_helpernode/helpernode.tf]: These are added for support external DNS/LB. Remove from them and will create a new commit for them.

**Comment by ltccci** [comment]
New changes are detected. LGTM label has been removed.

**Comment by yussufsh** [comment]
@cs-zhang @bpradipt see my comment

> Another thing, from code it is allowing me to optionally set only master fixed IPs, we could ignore worker,bootstrap fixed IPs. This will lead to the race condition and block the reserved IP. In other words, workers who can take dynamic IP may pick up the fixed IP for master from the subnet. See #182 where we faced similar issue.

**Comment by cs-zhang** [comment]
@yussufsh I already added the dependent of master ports for worker ports, that would solved the case you described.

**Comment by yussufsh** [comment]
> @yussufsh I already added the dependent of master ports for worker ports, that would solved the case you described.

Just wanted to make sure you have tested those. Thanks.

**Review by yussufsh — APPROVED** [review]

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/201#" title="Author self-approved">cs-zhang</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/201#pullrequestreview-653013382" title="LGTM">Prajyot-Parab</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/201#pullrequestreview-655900515" title="Approved">yussufsh</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervm).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/master/OWNERS)~~ [yussufsh]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
